src/middlewared/middlewared/api/v25_04_0/catalog.py

```diff
@@ -0,0 +1,102 @@
+
+from middlewared.api.base import BaseModel, ForUpdateMetaclass, NonEmptyString, single_argument_args
+
+
+__all__ = [
+    'CatalogEntry', 'CatalogUpdateArgs', 'CatalogUpdateResult', 'CatalogTrainsArgs', 'CatalogTrainsResult',
+    'CatalogSyncArgs', 'CatalogSyncResult', 'CatalogAppInfo', 'CatalogAppsArgs', 'CatalogAppsResult',
+    'CatalogAppDetailsArgs', 'CatalogAppDetailsResult',
+]
+
+
+class CatalogEntry(BaseModel):
+    id: NonEmptyString
+    label: NonEmptyString = Field(pattern=r'^\w+[\w.-]*$')
+    preferred_trains: list[NonEmptyString]
+
+
+@single_argument_args('catalog_update')
+class CatalogUpdateArgs(BaseModel, metaclass=ForUpdateMetaclass):
+    preferred_trains: list[NonEmptyString]
+
+
+class CatalogUpdateResult(BaseModel):
+    result: CatalogEntry
+
+
+class CatalogTrainsArgs(BaseModel):
+    pass
+
+
+class CatalogTrainsResult(BaseModel):
+    result: list[NonEmptyString]
+
+
+class CatalogSyncArgs(BaseModel):
+    pass
+
+
+class CatalogSyncResult(BaseModel):
+    result: None
+
+
+class Maintainer(BaseModel):
+    name: str
+    email: str
+    url: HttpUrl | None
+
+
+class CatalogAppInfo(BaseModel):
+    app_readme: str = Field(description='HTML content of the app README.')
+    categories: list[str] = Field(description='List of categories for the app.')
+    description: str = Field(description='Short description of the app.')
+    healthy: bool = Field(description='Health status of the app.')
+    healthy_error: str | None = Field(default=None, description='Error if app is not healthy.')
+    home: HttpUrl = Field(description='Homepage URL of the app.')
+    location: str = Field(description='Local path to the app\'s location.')
+    latest_version: str = Field(description='Latest available app version.')
+    latest_app_version: str = Field(description='Latest available app version in repository.')
+    latest_human_version: str = Field(description='Human-readable version of the app.')
+    last_update: str = Field(description='Timestamp of the last update in ISO format.')
+    name: str = Field(description='Name of the app.')
+    recommended: bool = Field(description='Indicates if the app is recommended.')
+    title: str = Field(description='Title of the app.')
+    maintainers: list[Maintainer] = Field(description='List of app maintainers.')
+    tags: list[str] = Field(description='Tags associated with the app.')
+    screenshots: list[HttpUrl] = Field(description='List of screenshot URLs.')
+    sources: list[HttpUrl] = Field(description='List of source URLs.')
+    icon_url: list[HttpUrl] = Field(None, description='URL of the app icon.')
+
+    # We do this because if we change anything in catalog.json, even older releases will
+    # get this new field and different roles will start breaking due to this
+    model_config = ConfigDict(extra='allow')
+
+
+@single_argument_args('catalog_apps_options')
+class CatalogAppsArgs(BaseModel):
+    cache: bool = Field(default=True)
```

Comment:
If you're only setting the default, it's enough to just have `cache: bool = True`.